eth_defi/aave_v3/events.py

```diff
@@ -0,0 +1,311 @@
+"""Aave v3 event reader.
+
+Efficiently read Aave v3 from a blockchain.
+
+Currently we are tracking these events:
+
+- ReserveDataUpdated
+"""
+import csv
+import datetime
+import logging
+from pathlib import Path
+
+from requests.adapters import HTTPAdapter
+from tqdm.auto import tqdm
+from web3 import Web3
+
+from eth_defi.aave_v3.constants import (
+    AAVE_V3_NETWORKS,
+    aave_v3_get_token_name_by_deposit_address,
+)
+from eth_defi.abi import get_contract
+from eth_defi.event_reader.conversion import (
+    convert_int256_bytes_to_int,
+    convert_uint256_string_to_address,
+    decode_data,
+)
+from eth_defi.event_reader.logresult import LogContext
+from eth_defi.event_reader.reader import LogResult, read_events_concurrent
+from eth_defi.event_reader.state import ScanState
+from eth_defi.event_reader.web3factory import TunedWeb3Factory
+from eth_defi.event_reader.web3worker import create_thread_pool_executor
+from eth_defi.token import TokenDetails, fetch_erc20_details
+
+# from eth_defi.token import TokenDetails, fetch_erc20_details
+
+logger = logging.getLogger(__name__)
+
+
+class TokenCache(LogContext):
+    """Manage cache of token data when doing ReserveDataUpdated look-up.
+
+    Do not do extra requests for already known tokens.
+    """
+
+    def __init__(self):
+        self.cache = {}
+
+    def get_token_info(self, web3: Web3, address: str) -> TokenDetails:
+        if address not in self.cache:
+            details = fetch_erc20_details(web3, address, raise_on_error=False)
+            logging.warn(f'Fetched ERC20 details for {address}: {details}')
```

Comment:
I suppose this should be `info` instead?
Fetching token details is a perfectly expected operation, and not something that we should be warned about?

Also, I believe it was meant to use `logger` instead?
(applies to multiple places)